### PR TITLE
Subscription timeouts should not hit after stream ref subscribe 

### LINKED ISSUE #24626: SourceRef throws RemoteStreamRefActorTerminatedException
I played around with the new StreamRefs and found the following problem. After a while (~30 seconds) I get the exception

```scala
Exception in thread "main" akka.stream.RemoteStreamRefActorTerminatedException: The remote partner Actor[akka://default/user/dataSource/StreamSupervisor-1/$$a-SinkRef-0#-49821885] has terminated! Tearing down this side of the stream as well.
	at akka.stream.impl.streamref.SourceRefStageImpl$$anon$1.$anonfun$initialReceive$1(SourceRefImpl.scala:176)
	at akka.stream.impl.streamref.SourceRefStageImpl$$anon$1.$anonfun$initialReceive$1$adapted(SourceRefImpl.scala:140)
	at akka.stream.stage.GraphStageLogic$StageActor.internalReceive(GraphStage.scala:230)
	at akka.stream.stage.GraphStageLogic$StageActor.$anonfun$callback$1(GraphStage.scala:198)
	at akka.stream.stage.GraphStageLogic$StageActor.$anonfun$callback$1$adapted(GraphStage.scala:198)
	at akka.stream.impl.fusing.GraphInterpreter.runAsyncInput(GraphInterpreter.scala:454)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:465)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:560)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:742)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:757)
	at akka.actor.Actor.aroundReceive(Actor.scala:517)
	at akka.actor.Actor.aroundReceive$(Actor.scala:515)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:667)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:590)
	at akka.actor.ActorCell.invoke(ActorCell.scala:559)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
``` 

Could it be that even though I materialize the SourceRef that I run in the subscription-timeout of 30 seconds?

```scala
import akka.NotUsed
import akka.actor.Actor
import akka.stream.{ActorMaterializer, SourceRef}
import akka.pattern.pipe
import akka.stream.scaladsl.{Source, StreamRefs}

import scala.concurrent.Future

case class RequestLogs(streamId: Int)
case class LogsOffer(streamId: Int, sourceRef: SourceRef[String])

class DataSource extends Actor {
  import context.dispatcher
  implicit val mat = ActorMaterializer()(context)

  def receive = {
    case RequestLogs(streamId) ⇒
      val source: Source[String, NotUsed] = streamLogs(streamId)
      val ref: Future[SourceRef[String]] =
        source.runWith(StreamRefs.sourceRef())
      val reply: Future[LogsOffer] = ref.map(LogsOffer(streamId, _))
      reply pipeTo sender()
  }

  def streamLogs(streamId: Long): Source[String, NotUsed] =
    Source.unfold(0)(i => Some((i + 1, s"Log message #$i")))
}
```

```scala
import akka.actor.{ActorSystem, Props}
import akka.pattern.ask
import akka.stream.scaladsl.Sink
import akka.stream.{ActorMaterializer, ThrottleMode}
import akka.util.Timeout

import scala.concurrent.Await
import scala.concurrent.duration._

object DataSourceApp extends App {
  implicit val system: ActorSystem = ActorSystem()
  implicit val mat: ActorMaterializer = ActorMaterializer()
  implicit val timeout: Timeout = Timeout(1 second)

  val sourceActor = system.actorOf(Props[DataSource], "dataSource")

  val offer = Await.result(sourceActor ? RequestLogs(1337), 1 second)

  val f = offer match {
    case LogsOffer(_, sourceRef) =>
      sourceRef
        .throttle(1, 1 second, 1, ThrottleMode.Shaping)
        .runWith(Sink.foreach(println))
    case e =>
      throw new Exception(s"Got unexpected message $e")
  }

  Await.result(f, Duration.Inf)
}
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #24626

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/streamref/SourceRefImpl.scala [hunk 1]
```diff
@@ -139,7 +139,7 @@ private[stream] final class SourceRefStageImpl[Out](
 
       lazy val initialReceive: ((ActorRef, Any)) ⇒ Unit = {
         case (sender, msg @ StreamRefsProtocol.OnSubscribeHandshake(remoteRef)) ⇒
-          cancelTimer("SubscriptionTimeoutTimerKey")
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/streamref/SourceRefImpl.scala]: D'oh!

Comment (ktoso) [reply akka-stream/src/main/scala/akka/stream/impl/streamref/SourceRefImpl.scala]: LOL, thanks

**Review by ktoso — APPROVED** [review]
LGTM, good catch

**Comment by ktoso** [comment]
Very obvious fix, merging -- thanks a lot Johan!

**Pull request merged**